var/ramble/repos/builtin/modifiers/nccl-env/modifier.py

```diff
@@ -0,0 +1,714 @@
+# Copyright 2022-2024 The Ramble Authors
+#
+# Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+# https://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+# <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
+# option. This file may not be copied, modified, or distributed
+# except according to those terms.
+
+from ramble.modkit import *  # noqa: F403
+import ramble.util.env
+import ramble.util.executable
+
+
+class NcclEnv(BasicModifier):
+    """Define a modifier for configuring NCCL based environment variables
+
+    NCCL is the NVIDIA Collective Communications Library, which provides
+    inter-GPU communication primitives that are topology-aware and can easily
+    be integrated into applications.
+
+    This modifier presents variables that can be use to configure and
+    parameterize aspects of NCCL's behavior. The available parameters are
+    documented in more details here:
+    https://docs.nvidia.com/deeplearning/nccl/user-guide/docs/env.html
+
+    Any variable who's value is an empty string will not be define in the
+    resulting execution script.
+    """
+
+    name = "nccl-env"
+
+    tags("gpu", "communication-performance")
+
+    maintainers("douglasjacobsen")
+
+    mode("standard", description="Standard execution mode")
+    default_mode("standard")
+
+    modifier_variable(
+        "nccl_socket_ifname",
+        default="",
+        description="Specification of which IP interfaces to use for communication",
+        modes=["standard"],
+    )
+
+    modifier_variable(
+        "nccl_socket_family",
+        default="",
+        description="Allows users to force IPv4 or IPv6 interfaces",
+        values=["AF_INET", "AF_INET6"],
+        modes=["standard"],
+    )
+
+    modifier_variable(
+        "nccl_socket_nthreads",
+        default="",
+        description="Specification of number of CPU helper threads used per network connection.",
+        modes=["standard"],
+    )
+
+    modifier_variable(
+        "nccl_nsocks_perthread",
+        default="",
+        description="Specification of number of sockets opened by each helper thread.",
+        modes=["standard"],
+    )
+
+    modifier_variable(
+        "nccl_cross_nic",
+        default="",
+        values=[0, 1, 2],
```

Comment:
Does this essentially make it a required variable?

Reply:
No, this just allows `ramble info --type modifiers nccl-env` to print "suggested values" for this variable. It's more for documentation than any kind of enforcement.